cmd/rpcfuzz/rpcfuzz.go

```diff
@@ -0,0 +1,1395 @@
+		Address   string        `json:"address,omitempty"`
+		Topics    []interface{} `json:"topics,omitempty"`
+	}
+)
+
+const (
+	codeQualityPrivateKey = "42b6e34dc21598a807dc19d7784c71b2a7a01f6480dc6f58258f78e539f1a1fa"
+
+	defaultGas                  = "0x100000"
+	defaultGasPrice             = "0x1000000000"
+	defaultMaxFeePerGas         = "0x1000000000"
+	defaultMaxPriorityFeePerGas = "0x1000000000"
+)
+
+var (
+	testPrivateHexKey     *string
+	testContractAddress   *string
+	testPrivateKey        *ecdsa.PrivateKey
+	testEthAddress        ethcommon.Address
+	testNamespaces        *string
+	testAccountNonce      uint64
+	testAccountNonceMutex sync.Mutex
+	currentChainID        *big.Int
+
+	enabledNamespaces []string
+
+	// in the future allTests could be used to for
+	// fuzzing.. E.g. loop over the various tests, and mutate the
+	// Args before sending
+	allTests = make([]RPCTest, 0)
+)
+
+// setupTests will add all of the `RPCTests` to the `allTests` slice.
+func setupTests(ctx context.Context, rpcClient *rpc.Client) {
+
+	// cast rpc --rpc-url localhost:8545 net_version
+	allTests = append(allTests, &RPCTestGeneric{
+		Name:      "RPCTestNetVersion",
+		Method:    "net_version",
+		Args:      []interface{}{},
+		Validator: ValidateRegexString(`^\d*$`),
+	})
+
+	// cast rpc --rpc-url localhost:8545 web3_clientVersion
+	allTests = append(allTests, &RPCTestGeneric{
+		Name:      "RPCTestWeb3ClientVersion",
+		Method:    "web3_clientVersion",
+		Args:      []interface{}{},
+		Validator: ValidateRegexString(`^[[:print:]]*$`),
+	})
+
+	// cast rpc --rpc-url localhost:8545 web3_sha3 0x68656c6c6f20776f726c64
+	allTests = append(allTests, &RPCTestGeneric{
+		Name:      "RPCTestWeb3SHA3",
+		Method:    "web3_sha3",
+		Args:      []interface{}{"0x68656c6c6f20776f726c64"},
+		Validator: ValidateRegexString(`0x47173285a8d7341e5e972fc677286384f802f8ef42a5ec5f03bbfa254cb01fad`),
+	})
+
+	allTests = append(allTests, &RPCTestGeneric{
+		Name:      "RPCTestWeb3SHA3Error",
+		IsError:   true,
+		Method:    "web3_sha3",
+		Args:      []interface{}{"68656c6c6f20776f726c64"},
+		Validator: ValidateError(`cannot unmarshal hex string without 0x prefix`),
+	})
+
+	// cast rpc --rpc-url localhost:8545 net_listening
+	allTests = append(allTests, &RPCTestGeneric{
+		Name:      "RPCTestNetListening",
+		Method:    "net_listening",
+		Args:      []interface{}{},
+		Validator: ValidateExact(true),
+	})
+
+	// cast rpc --rpc-url localhost:8545 net_peerCount
+	allTests = append(allTests, &RPCTestGeneric{
+		Name:      "RPCTestNetPeerCount",
+		Method:    "net_peerCount",
+		Args:      []interface{}{},
+		Validator: ValidateRegexString(`^0x[[:xdigit:]]*$`),
+	})
+
+	// cast rpc --rpc-url localhost:8545 eth_protocolVersion
+	allTests = append(allTests, &RPCTestGeneric{
+		Name:      "RPCTestEthProtocolVersion",
+		IsError:   true,
+		Method:    "eth_protocolVersion",
+		Args:      []interface{}{},
+		Validator: ValidateError(`method eth_protocolVersion does not exist`),
```

Comment:
nit and maybe something for me to pickup. regex matches might be risky here as there doesn't seem to be a clear consensus on what error message to return (even in the spec [doc](https://www.jsonrpc.org/specification) examples showing different error message). safer way would be to verify for a JSON RPC status code of -32601.

Reply:
Interesting... Yeah my guess is these will fail across different clients. As we start to test and understand the ways in which they vary, I think we'll be in a good spot to generalize.